src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp

```diff
@@ -46,5 +70,142 @@ bool troubled_cell_indicator(
                                gsl::span<std::pair<size_t, size_t>>>,
                      VolumeDim>& volume_and_slice_indices) noexcept;
 
+template <size_t VolumeDim, typename PackagedData>
+DirectionMap<VolumeDim, double> compute_effective_neighbor_sizes(
+    const Element<VolumeDim>& element,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data) noexcept {
+  DirectionMap<VolumeDim, double> result;
+  for (const auto& dir : Direction<VolumeDim>::all_directions()) {
+    const auto& externals = element.external_boundaries();
+    const bool neighbors_in_this_dir = (externals.find(dir) == externals.end());
+    if (neighbors_in_this_dir) {
+      const double effective_neighbor_size =
+          [&dir, &element, &neighbor_data ]() noexcept {
+        const size_t dim = dir.dimension();
+        const auto& neighbor_ids = element.neighbors().at(dir).ids();
+        double size_accumulate = 0.;
+        for (const auto& id : neighbor_ids) {
+          size_accumulate += gsl::at(
+              neighbor_data.at(std::make_pair(dir, id)).element_size, dim);
+        }
+        return size_accumulate / neighbor_ids.size();
+      }
+      ();
+      result.insert(std::make_pair(dir, effective_neighbor_size));
+    }
+  }
+  return result;
+}
+
+template <typename Tag, size_t VolumeDim, typename PackagedData>
+DirectionMap<VolumeDim, double> compute_effective_neighbor_means(
+    const Element<VolumeDim>& element, const size_t tensor_storage_index,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data) noexcept {
+  DirectionMap<VolumeDim, double> result;
+  for (const auto& dir : Direction<VolumeDim>::all_directions()) {
+    const auto& externals = element.external_boundaries();
+    const bool neighbors_in_this_dir = (externals.find(dir) == externals.end());
+    if (neighbors_in_this_dir) {
+      const double effective_neighbor_mean =
+          [&dir, &element, &neighbor_data, &tensor_storage_index ]() noexcept {
+        const auto& neighbor_ids = element.neighbors().at(dir).ids();
+        double mean_accumulate = 0.0;
+        for (const auto& id : neighbor_ids) {
+          mean_accumulate +=
+              get<::Tags::Mean<Tag>>(neighbor_data.at(std::make_pair(dir, id))
+                                         .means)[tensor_storage_index];
+        }
+        return mean_accumulate / neighbor_ids.size();
+      }
+      ();
+      result.insert(std::make_pair(dir, effective_neighbor_mean));
+    }
+  }
+  return result;
+}
```

Comment:
These three helper functions (`allocate_buffers`, `compute_effective_neighbor_sizes`, `compute_effective_neighbor_means`) don't need to be in the MinmodTci file. I will move them into a MinmodHelpers file, for a cleaner organization.